lib/fileset/tests.sh

```diff
@@ -894,6 +894,37 @@ expectEqual \
     'toSource { root = ./.; fileset = union ./d/a ./d/b; }'
 rm -rf -- *
 
+# Check that .hasExt checks for the file extension
+# The empty extension is the same as a file ending with a .
+tree=(
+    [a]=0
+    [a.]=1
+    [a.b]=0
+    [a.b.]=1
+    [a.b.c]=0
+)
+checkFileset 'fileFilter (file: file.hasExt "") ./.'
+# It can check for the last extension
+tree=(
+    [a]=0
+    [.a]=1
+    [.a.]=0
+    [.b.a]=1
+    [.b.a.]=0
+)
+checkFileset 'fileFilter (file: file.hasExt "a") ./.'
+# It can check for any extension
+tree=(
+    [a.b.c.d]=1
+)
+checkFileset 'fileFilter (file:
+  all file.hasExt [
+    "b.c.d"
+    "c.d"
+    "d"
+  ]
+) ./.'
+
```

Comment:
Does `.gitignore` have a `.gitignore` extension, or no extension at all? In other words, does the empty basename exist?

Reply:
This is covered with the `file.hasExt "a"` test! Should be added to the docs though :)

Reply:
I guess we could ask whether that's the right behavior though, alternatively it could error :thinking:

Reply:
I think it's fine to leave it as is, I added some docs for that: [(diff)](https://github.com/NixOS/nixpkgs/compare/5aaac6d99bc48db19916e5e8f2760694a55169cf..6816f28c960c523e6a30f2ad4a1cc812251f5ffb)